tests/nat/nat_helpers.py

```diff
@@ -0,0 +1,848 @@
+        return setup_info[interface_type]['inner_port_id']
+    return setup_info[interface_type]['outer_port_id']
+
+
+def get_src_ip(setup_info, direction, interface_type, nat_type=None, second_port=False):
+    """
+    return source IP  based on test case direction and interface_type
+    :param setup_info: setup info fixture
+    :param direction: 'host-tor', 'leaf-tor'
+    :param interface_type: type of interface
+    :param second_port: boolean if second port's IP settings need to be returned
+    :param nat_type: string nat type
+    :return: source IP
+    """
+    if direction == 'host-tor' or nat_type == "static_napt":
+        if second_port:
+            return setup_info[interface_type]["second_src_ip"]
+        return setup_info[interface_type]['src_ip']
+    return setup_info[interface_type]['dst_ip']
+
+
+def get_dst_ip(setup_info, direction, interface_type, nat_type=None, second_port=False):
+    """
+    return destination IP  based on test case direction and interface_type
+    :param setup_info: setup info fixture
+    :param direction: 'host-tor', 'leaf-tor'
+    :param interface_type: type of interface
+    :param second_port: boolean if second port's IP settings need to be returned
+    :param nat_type: string nat type
+    :return: destination IP
+    """
+    if direction == 'host-tor' or nat_type == "static_napt":
+        return setup_info[interface_type]['dst_ip']
+    return setup_info[interface_type]['public_ip']
+
+
+def get_public_ip(setup_info, interface_type, second_port=False):
+    """
+    return public IP  based on test case interface_type
+    :param setup_info: setup info fixture
+    :param interface_type: type of interface
+    :param second_port: boolean if second port's IP settings need to be returned
+    :return: public IP
+    """
+    if second_port and interface_type == "port_in_lag":
+        return setup_info[interface_type]['second_public_ip']
+    return setup_info[interface_type]['public_ip']
+
+
+def setup_ptf_interfaces(ptfhost, duthost, setup_info, interface_type, vrf_id, vrf_name, port_id,
+                         ip_address, mask, gw_ip, key):
+    """
+    setup ptf interfaces for tests
+    :param ptfhost: PTF host object
+    :param duthost: DUT host object
+    :param setup_info: setup info fixture
+    :param interface_type: string interface type
+    :param vrf_id: id of vrf
+    :param vrf_name: vrf name
+    :param port_id: port id of interface
+    :param ip_address: ip address of interface
+    :param mask: vrf mask
+    :param gw_ip: ip address of gateway
+    :param key: dictionary key if vrf configuration
+    """
+    ptfhost.shell("grep -Fxq '{} {}' /etc/iproute2/rt_tables "
+                  "|| echo '{} {}' >> /etc/iproute2/rt_tables".format(vrf_id, vrf_name, vrf_id, vrf_name))
+    ptfhost.shell("ip link add {} type vrf table {}".format(vrf_name, vrf_id))
+    ptfhost.shell("ip link set dev {} up".format(vrf_name))
```

Comment:
why do we need the vrf in the ptf docker

Reply:
Each NAT test case needs client/server traffic transmission.
NAT starts addresses/ports translation only if connection tracking is taking place(i.e. for TCP handshake).

PTF performs traffic transmission as a client and as a server at the same time.
That approach needs testbed be customized to:

 * avoid limitaion of PTFs injected interfaces(i.e. traffic on DUT's ingress from VMs cannot be captured)
 * keep PTF's client/server traffic isolated (using VRFs)

The customized testbed with applied T0 topo for each NAT test case looks as follows:

![image](https://user-images.githubusercontent.com/25028638/88930505-3ed6c500-d284-11ea-9a4a-c2b968732444.png)

After end of the test session teardown procedure turns testbed to the initial state.